### PR TITLE
docs: add security response process

### PR DESCRIPTION
**Pull request by moolen**
This PR proposes a security incident response process / guideline just to have it in case.

This is taken from the [CNC/tag-security Project Resources](https://github.com/cncf/tag-security/blob/231b87f371274b2d68def2c6a35a719210836191/project-resources/README.md). Minor adaptations have been made.

### CODE REVIEW

File: SECURITY_RESPONSE.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+confirmation occurs, providing the notification, and issuing a security advisory
+as well as patch/release.
+
+## Triage
+
+### Identify the problem
+
+Triaging problems allows maintainers to focus resources on the most critically
+impacting problems. Potential security problems should be evaluated against the
+following information:
```
Comment (Skarlso) [inline SECURITY_RESPONSE.md:13]:
```suggestion
Triaging issues allows maintainers to focus resources on the most critically
impacting problems. Potential security risks should be evaluated against the
following information:
```

File: SECURITY_RESPONSE.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+* What privileges are required for this problem to occur?
+  * admin
+  * general
+* What is the potential impact or consequence of the problem?
+* Does an exploit exist?
+
+Any potential problem that has an exploit, permits privilege escalation, is
+simple, and does not require user interaction should be evaluated immediately.
+[CVSS Version 3.1](https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator) can be
+a helpful tool in evaluating the criticality of reported problems.
```
Comment (Skarlso) [inline SECURITY_RESPONSE.md:33]:
```suggestion
a helpful tool in evaluating the criticality of reported issues.
```

File: SECURITY_RESPONSE.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+### Acknowledge receipt of the problem
+
+Respond to the reporter and notify them you have received the problem and have
+begun reviewing it. Remind them of the [embargo policy](https://github.com/cncf/tag-security/blob/231b87f371274b2d68def2c6a35a719210836191/project-resources/templates/embargo-policy.md), and provide them
+information on who to contact/follow-up with if they have questions. Estimate a
+time frame that they can expect to receive an update on the problem. Create a
+calendar reminder to contact them again by that date to provide an update.
```
Comment (Skarlso) [inline SECURITY_RESPONSE.md:41]:
```suggestion
Respond to the reporter and notify them that you have received and begun reviewing the problem. Remind them of the [embargo policy](https://github.com/cncf/tag-security/blob/231b87f371274b2d68def2c6a35a719210836191/project-resources/templates/embargo-policy.md), and provide them
information on who to contact/follow-up with if they have questions. Estimate when they can expect to receive an update. Create a calendar reminder to contact them again by that date to provide an update.
```

Comment (Skarlso) [reply SECURITY_RESPONSE.md:41]: uh I think I messed up the linebreaks a bit here...

**Review by Skarlso — APPROVED** [review]
LGTM, just have a couple of suggestions for removing duplicate words and adding a bit of clarity.

**Comment by moolen** [comment]
Thanks! I've opened https://github.com/external-secrets/external-secrets/pull/3056 to get your suggestions in!

**Pull request merged**